plugin/init.lua

```diff
@@ -105,24 +114,30 @@ pub.encryption = {
 	private_key = nil,
 	public_key = nil,
 	encrypt = function(file_path, lines)
+		wezterm.emit("resurrect.encrypt.start", file_path)
 		local cmd = string.format("age -r %s -o %s", pub.encryption.public_key, file_path:gsub(" ", "\\ "))
 		local success, output = execute_cmd_with_stdin(cmd, lines)
 
 		if not success then
+			wezterm.emit("resurrect.error", "Encryption failed: " .. output)
 			wezterm.log_error("Encryption failed: " .. output)
 			return
 		end
+		wezterm.emit("resurrect.encrypt.finished", file_path)
 	end,
 	decrypt = function(file_path)
+		wezterm.emit("resurrect.decrypt.start", file_path)
 		local success, stdout, stderr =
 			wezterm.run_child_process({ "age", "-d", "-i", pub.encryption.private_key, file_path })
 		if not success then
+			wezterm.emit("resurrect.error", "Decryption failed: " .. stderr)
 			wezterm.log_error("Decryption failed: " .. stderr)
 			return
```

Comment:
Would it be better to error here instead of return and then handle the error in the function which calls this?

Reply:
Would you prefer to move the emit error logging from `encrypt` and `decrypt` to the `save_state()` and `load_state()` functions?

Reply:
Yes, I think that is better, because if a user want to provide their own encryption/decryption functions, then errors would also be supported for user provided functions.

Reply:
Moved in latest push